pkg/cli/mt_start_sql.go

```diff
@@ -62,19 +62,16 @@ func runStartSQL(cmd *cobra.Command, args []string) error {
 		if err := serverCfg.InitSQLServer(ctx); err != nil {
 			return err
 		}
-		// This value is injected in order to have something populated during startup.
-		// In the initial 20.2 release of multi-tenant clusters, no version state was
-		// ever populated in the version cluster setting. A value is populated during
-		// the activation of 21.1. See the documentation attached to the TenantCluster
-		// in upgrade/upgradecluster for more details on the tenant upgrade flow.
-		// Note that the value of 21.1 is populated when a tenant cluster is created
-		// during 21.1 in crdb_internal.create_tenant.
-		//
-		// Note that the tenant will read the value in the system.settings table
-		// before accepting SQL connections.
+
+		// We need a value in the version setting prior to the update
+		// coming from the system.settings table. This value must be valid
+		// and compatible with the state of the tenant's keyspace.
 		//
-		// TODO(knz): Check if this special initialization can be removed.
-		// See: https://github.com/cockroachdb/cockroach/issues/90831
+		// Since we don't know at which binary version the tenant
```

Comment:
This is a bit pedantic, but I don't agree that assuming an old binary version is the conservative choice. Some versions are one way doors and using an old version after the migration runs is also broken. 

The pain point here is we don't have a way to represent "unknown version". The tenant needs an up to date setting before any version gates are checked that make durable decisions. 

If anything, picking the oldest version is actually more likely to cause problems than picking the newest version, since most clusters are running with binary version == cluster version. But that is probably a good thing, because it increases the probability we will catch version bugs during initialization.

Reply:
I'm going to merge this but we should leave this convo open. I hope that @healthy-pod can weigh in when he's back.

Reply:
I see two questions here:
1. Can the current state lead to issues?
Hopefully not, because we won't run any migrations/upgrades or check version gates before starting the settings watcher. When the settings watcher is started in `SQLServer.preStart`, it will set the right version and we will be good.

2. Do we need to initialize the version where it gets initialized now? Instead of that we may be able to initialize it in `preStart` with the right version (tenant logical version).
I don't have an answer yet. Do you have any thoughts on this?

Reply:
> Do we need to initialize the version where it gets initialized now? Instead of that we may be able to initialize it in preStart with the right version (tenant logical version).
> I don't have an answer yet. Do you have any thoughts on this?

I think it would be a good thing to investigate. Mind filing an issue?

Reply:
https://github.com/cockroachdb/cockroach/pull/104859